sdks/go/node/api/client/embedLocalFilesAndDirs.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"fmt"
+	"github.com/opctl/opctl/sdks/go/model"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+//embedLocalFilesAndDirs mutates args by embedding any referenced file &/or dirs as objects.
+// This makes the args location independent and therefore runnable on a remote node.
+// note1: this loads referenced files/dirs into memory so can easily exhaust available memory.
+// note2: this approach is primitive; we will need to support de-dupe and chunking in the future.
+func embedLocalFilesAndDirs(
+	args map[string]*model.Value,
```

Comment:
does this need to be a ptr, since this func doesnt return anything?
```
args *map[string]*model.Value
```

Reply:
no because maps in golang are reference types

Reply:
gotcha